### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by vsupruniuk**

### CODE REVIEW

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,25 @@
...
+        delete state[actions[i].keysToRemove[key]];
+      }
+    }
+
+    if (actions[i].type === 'clear') {
+      for (const key in state) {
+        delete state[key];
+      }
+    }
+  }
```
Comment (Olexandr6) [inline src/transformState.js:26]: [Checklist](https://github.com/mate-academy/js_stateful-object/blob/master/checklist.md) 2, 4 points

**Review by G-YaSha — CHANGES_REQUESTED** [review]

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,30 @@
...
+      case 'addProperties':
+        for (const key in actions[i].extraData) {
+          state[key] = actions[i].extraData[key];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const key in actions[i].keysToRemove) {
+          delete state[actions[i].keysToRemove[key]];
+        }
```
Comment (illiarerun) [inline src/transformState.js:19]: [Checklist](https://github.com/mate-academy/js_stateful-object/blob/master/checklist.md) 1st item, don't use `for ... in` for array iteration

File: src/transformState.js [hunk 3]
```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        for (const key in actions[i].extraData) {
+          state[key] = actions[i].extraData[key];
+        }
```
Comment (illiarerun) [inline src/transformState.js:13]: You can do it with Object methoed designed for `assigning` values from other object

**Review by illiarerun — CHANGES_REQUESTED** [review]

File: src/transformState.js [hunk 4]
```diff
@@ -5,7 +5,28 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const value of actions) {
```
Comment (illiarerun) [inline src/transformState.js:8]: come on, [checklist](https://github.com/mate-academy/js_stateful-object/blob/master/checklist.md) 3rd item, use proper variable names when iterating through array using `for ... of`
```suggestion
  for (const action of actions) {
```

**Review by illiarerun — APPROVED** [review]
GJ, left one comment, if I see this in next task I'll reject))

**Pull request closed without merging**